### PR TITLE
Use source level in FunctionPackage

### LINKED ISSUE #148: Java 8 code generated in javac 8 when source level is set to 7
This is due to the workaround we have for an Eclipse bug (namely, it never sets source level > 7). We should somehow check to see if we're in Eclipse, and check source level if we're not.

### PR DESCRIPTION
**Pull request by chrisalice**
Eclipse is bugged, and will never return source level > 7. Previously we hacked around this by only checking for the presence of a java.util.function type in the processing environment, but this means we generate Java 8 code in javac 8 even when source level is set to 6/7 unless users set the bootstrap classpath correctly. Now we check for the presence of the Eclipse round environment on the stack trace, and only skip the source level check if we find it.

This fixes #148.

### CODE REVIEW

File: src/main/java/org/inferred/freebuilder/processor/util/feature/FunctionPackage.java [hunk 1]
```diff
@@ -30,12 +32,18 @@ protected FunctionPackage testDefault() {
 
         @Override
         protected FunctionPackage forEnvironment(ProcessingEnvironment env) {
-          TypeElement element = env.getElementUtils()
-              .getTypeElement(UNARY_OPERATOR.getQualifiedName().toString());
-          return (element != null) ? AVAILABLE : UNAVAILABLE;
+          if (runningInEclipse()) {
+            // Eclipse is bugged: sourceVersion will never be > 7.
+            // Work around this by checking for the presence of java.util.function.Consumer instead.
+            return hasType(env.getElementUtils(), CONSUMER) ? AVAILABLE : UNAVAILABLE;
```
Comment (j-baker) [inline src/main/java/org/inferred/freebuilder/processor/util/feature/FunctionPackage.java]: why does this not suffer from the problem of compiling in Eclipse 1.8 for Java 1.7? Or is this just a tradeoff that you'll have to make whatever way?

**Comment by j-baker** [comment]
lgtm

File: src/main/java/org/inferred/freebuilder/processor/util/feature/FunctionPackage.java (hunk 1 above)
Comment (chrisalice) [inline src/main/java/org/inferred/freebuilder/processor/util/feature/FunctionPackage.java]: Yeah, tradeoff. It's easier to configure Eclipse to use the correct compiler, too.

Comment (j-baker) [inline src/main/java/org/inferred/freebuilder/processor/util/feature/FunctionPackage.java]: cool

**Pull request merged**